configuration_go/schemas/thanos/cache/cache.go

```diff
@@ -0,0 +1,17 @@
+package cache
+
+type IndexCacheProvider string
+
+// Taken from github.com/thanos-io/pkg/store/cache/factory.go
+
+const (
+	INMEMORY  IndexCacheProvider = "IN-MEMORY"
+	MEMCACHED IndexCacheProvider = "MEMCACHED"
+	REDIS     IndexCacheProvider = "REDIS"
+)
```

Comment:
Are these used? 🤔

Reply:
What do you mean?
These are the supported types of caches supported by Thanos, extracted from Thanos. So, when you want to set the config as a flag content for a given component of Thanos, you can use the `IndexCacheConfig` and serialize it to yaml.

By the way, the reason all this code is copy pasted form Thanos and not directly imported is because `yaml` struct tags lack the `omitempty` tag and some of these struct also have additional fields not related to the input.